crates/uv/tests/tool_upgrade.rs

```diff
@@ -0,0 +1,188 @@
+#![cfg(all(feature = "python", feature = "pypi"))]
+
+use assert_fs::prelude::*;
+
+use common::{uv_snapshot, TestContext};
+
+mod common;
+
+#[test]
+fn test_tool_upgrade_name() {
+    let context = TestContext::new("3.12")
+        .with_filtered_counts()
+        .with_filtered_exe_suffix();
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    // Install `black`
+    uv_snapshot!(context.filters(), context.tool_install()
+        .arg("black==23.1")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .env("PATH", bin_dir.as_os_str()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv tool install` is experimental and may change without warning.
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     + black==23.1.0
+     + click==8.1.7
+     + mypy-extensions==1.0.0
+     + packaging==24.0
+     + pathspec==0.12.1
+     + platformdirs==4.2.0
+    Installed 2 executables: black, blackd
+    "###);
+
+    // Upgrade `black`
+    uv_snapshot!(context.filters(), context.tool_upgrade()
+        .arg("black")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .env("PATH", bin_dir.as_os_str()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv tool upgrade` is experimental and may change without warning.
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Uninstalled [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     - black==23.1.0
+     + black==24.3.0
+    Installed 2 executables: black, blackd
```

Comment:
Should these say "Updated"?

Reply:
```
Installed [N] packages in [TIME]
 - black==23.1.0
 + black==24.3.0
Installed 3 executables:
 ^ black
 ^ blackd
 + added-exe
 - removed-exe
```
same message format as package (un)installations diffs.
`^` mark stands for _Updated_ executable.

Reply:
> `^` mark stands for _Updated_ executable.


As I'm thinking now, could we have this _Updated_ versions in package installations? example:
```
Installed [N] packages in [TIME]
 ^ black 23.1.0 -> 24.3.0
```

Reply:
I've gone with `Updated` for now. Once a symbol is settled in #5213 for `updated`, we can update this. Wdyt?